src/components/common/dialog.tsx

```diff
@@ -0,0 +1,1550 @@
+     */
+
+    #trigger_validation(): void {
+        this.#validation = { };
+        this.#validation_failed = false;
+        this.#validate_callback(this);
+        this.#update();
+    }
+
+    #set_validation(path: string, result: string | undefined) {
+        if (result) {
+            this.#validation[path] = result;
+            this.#validation_failed = true;
+            this.#online_validation = true;
+            this.#update();
+        }
+    }
+
+    _get_validation(path: string): string | undefined {
+        if (path in this.#validation)
+            return this.#validation[path];
+        else
+            return undefined;
+    }
+
+    /* In between #trigger_validation and #set_validation, a lot is
+       going on, especially with asynchronous validation.
+
+       We use a DialogValidationState object to keep the necessary
+       state for that, such as cached results, and timeouts and
+       promises.
+
+       Note that a DialogValidationState object can change which path
+       it is for, see _rename_validation_state below. So we have to be
+       careful to always get the path out of the DialogValidationState
+       object.
+     */
+
+    #get_validation_state(path: string): DialogValidationState {
+        if (!(path in this.#validation_state))
+            this.#validation_state[path] = {
+                path,
+                cached_value: undefined,
+                cached_result: undefined,
+                timeout_id: 0,
+                promise: undefined,
+                round_id: undefined,
+            };
+        return this.#validation_state[path];
+    }
+
+    /* Calling #set_validation_state_result is the final thing that
+       should happen when validating a given path. It will install the
+       result in the cache and then call #set_validation.
+     */
+
+    #set_validation_state_result(
+        state: DialogValidationState,
+        val: unknown,
+        result: string | undefined,
+    ) {
+        state.cached_value = val;
+        state.cached_result = result;
+        state.timeout_id = 0;
+        state.promise = undefined;
+        state.round_id = undefined;
+        this.#set_validation(state.path, result);
+    }
+
+    /* The first thing should be of course to probe that cache.  If we
+       get a hit, it is used immediately to call #set_validation.
+
+       In that case, the DialogValidationState is also made part of
+       the current round since any asynchronous validation that is
+       currently running is still relevant. See below for more about
+       that.
+     */
+
+    #probe_validation_state_cache(state: DialogValidationState, val: unknown): boolean {
+        if (Object.is(state.cached_value, val)) {
+            state.round_id = this.#get_current_validation_round_id();
+            debug("cache hit", state.path, state.cached_result);
+            this.#set_validation(state.path, state.cached_result);
+            return true;
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2449-7ce1b174-20251217-183650-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)